Update all dependencies.

cansin opened the pull request:
##### *To:*
@jjinux cc: @udemy/team-f 

##### *What:*
I am upgrading every external dependency these packages have to the latest. It spawned out from my Webpack 4 upgrade work. Most notably,

1. It moves us off of `-beta` versions of babel into the latest, with minimal test-case change.
2. The latest `eslint` version, unfortunately, does not support node 8.9.1. Therefore we'd need to first upgrade node to `"node": "^6.14.0 || ^8.10.0 || >=9.10.0"` (i.e. either latest 8.x `8.12.0`, or latest LTS `10.13.0`). I ran the tests with both node versions and they are passing.

I will keep this PR here and perhaps ping SRE to see if they have the bandwidth for a node upgrade. Wdyt?

##### *JIRA:*
https://udemyjira.atlassian.net/browse/TF-3646

##### *What did you test:*
Run `yarn test`.

##### *What dashboards will you be monitoring:*
None.

Commit:
Update all dependencies.

cansin commented:
I am trying to figure out how I can end up with a docker image that would have both node >8.10 and yarn 1.9.4

cansin commented:
Ah, look at that! The default `node:8.12` docker repo comes with `yarn:1.9.4`, nice.

cansin commented on .circleci/config.yml:
```diff
@@ -6,26 +6,24 @@ version: 2
       - restore_cache:
           keys:
-          - v2-dependencies-{{ checksum "yarn.lock" }}
+          - v3-dependencies-{{ checksum "yarn.lock" }}
```
This v2 -> v3 change was probably unnecessary.

cansin commented on packages/babel-preset-udemy-website/raw.js:
```diff
@@ -37,7 +37,7 @@ module.exports = {
         // There are some caveats as to how to set this up:
         // https://new.babeljs.io/docs/en/next/babel-plugin-proposal-decorators.html
-        ['@babel/plugin-proposal-decorators', { legacy: true, decoratorsBeforeExport: true }],
+        ['@babel/plugin-proposal-decorators', { legacy: true }],
```
The babel no longer allows `legacy` and `decoratorsBeforeExport` at the same time. My understanding is `legacy` is already `decoratorsBeforeExport=true`.

cansin commented on packages/babel-preset-udemy-website/tests/result/subclassing-jsx-decorators-initializers.js:
```diff
@@ -144,15 +145,18 @@ function (_Component) {
     }
 
-    return babelHelpers.possibleConstructorReturn(_this, (_temp = _this = babelHelpers.possibleConstructorReturn(this, (_babelHelpers$getProt = babelHelpers.getPrototypeOf(Class)).call.apply(_babelHelpers$getProt, [this].concat(args))), babelHelpers.initializerDefineProperty(_this, "prop1", _descriptor, babelHelpers.assertThisInitialized(babelHelpers.assertThisInitialized(_this))), _this.prop2 =  ...
+    _this = babelHelpers.possibleConstructorReturn(this, (_babelHelpers$getProt = babelHelpers.getPrototypeOf(Class)).call.apply(_babelHelpers$getProt, [this].concat(args)));
```
I am not sure what to make of this change. It looks innocent enough.

jjinux replied:
Yeah, when you upgrade Babel, things like this change. You can glance at it, but you don't have to worry too much.

cansin commented:
I pinged @scott-estes and asked when would be a good time to do a 8.9.1 to 8.12.0 node upgrade.

jjinux commented on .circleci/config.yml:
```diff
@@ -6,26 +6,24 @@ version: 2
   build:
     docker:
-      - image: circleci/node:8.9
+      - image: node:8.12.0
```
Do you want to cut this PR in half and ship the parts that don't require the Node upgrade before you get the Node upgrade?

cansin replied:
I just downgraded eslint to a version that works with node 8.9.1

cansin replied:
This is deliberately left as node:8.12.0 as the other image is shipped with yarn 1.3.2. I figured having a forward minor version of node is less destructive than having the wrong version of yarn.

jjinux reviewed (approved):
LGTM

Commit:
Downgrade eslint to let the PR work with our current node version (8.9.1)

Commit:
node:8.9.1 image ships with yarn:1.3.2. We need yarn:1.9.4

Therefore I am upgrading docker image to node:8.12.0. We are going to upgrade to node:8.12.0 in a matter of weeks, therefore I do not think this is a problem.
